File: app/src/main/java/ru/visualmath/android/api/model/Lecture.java
package ru.visualmath.android.api.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;
import java.util.Locale;

public class Lecture implements Serializable {

    @SerializedName("_id")
    private String id;

    @SerializedName("name")
    private String name;

    @SerializedName("mapping")
    private List<Page> mapping;

    @SerializedName("modules")
    private List<Module> modules;

    @SerializedName("questions")
    private List<Question> questions;

    @SerializedName("questionBlocks")
    private List<String> questionBlocks;

    @SerializedName("hidden")
    private boolean hidden;

    @SerializedName("created")
    private String created;

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public List<Page> getMapping() {
        return mapping;
    }

    public List<Module> getModules() {
        return modules;
    }

    public List<Question> getQuestions() {
        return questions;
    }

    public List<String> getQuestionBlocks() {
        return questionBlocks;
    }

    public boolean isHidden() {
        return hidden;
    }

    public Date getCreatedDate() {
        DateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS", Locale.getDefault());
        Date date = new Date(0);
        try {
            date = format.parse(created);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return date;
    }
}

File: app/src/main/java/ru/visualmath/android/api/model/Lecture.kt
package ru.visualmath.android.api.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

@Parcelize
data class Lecture(

        @SerializedName("_id")
        val id: String = "",

        @SerializedName("name")
        val name: String = "",

        @SerializedName("mapping")
        val mapping: List<Page> = emptyList(),

        @SerializedName("modules")
        val modules: List<Module> = emptyList(),

        @SerializedName("questions")
        val questions: List<Question> = emptyList(),

        @SerializedName("questionBlocks")
        val questionBlocks: List<String> = emptyList(),

        @SerializedName("hidden")
        val isHidden: Boolean = false,

        @SerializedName("created")
        private val created: String = ""
) : Parcelable {

    val createdDate: Date
        get() {
            val format = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS", Locale.getDefault())
            var date = Date(0)
            try {
                date = format.parse(created)
            } catch (e: ParseException) {
                e.printStackTrace()
            }

            return date
        }
}
